### PR TITLE
MULE-12903: Every exported Mule package must contain api (part2)

### PR DESCRIPTION
**Pull request by pablokraan**
_ Moving org.mule.runtime.core.util.xa to internal/api
_ Moving org.mule.runtime.core.util.store to internal/api
_ Moving org.mule.runtime.core.util.queue to internal/api

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/privileged/util/queue/QueueManagerUtils.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.mule.runtime.core.api.util.FileUtils;
+import org.mule.runtime.core.internal.util.queue.DualRandomAccessFileQueueStoreDelegate;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Provides utilities to work with queues for testing purposes without having to export all the classes in
+ * org.mule.runtime.core.internal.util.queue package. For testing purposes only.
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/privileged/util/queue/QueueManagerUtils.java]: put the package as `@link` or `@code`

File: core/src/main/java/org/mule/runtime/core/internal/util/queue/QueueFileProvider.java [hunk 2]
```diff
@@ -22,8 +22,9 @@
  */
 public class QueueFileProvider {
 
-  public static final String OPEN_FILE_OPTIONS = "rws";
-  public static final String UTF_8_ENCODING = "UTF-8";
+  private static final String OPEN_FILE_OPTIONS = "rws";
+  private static final String UTF_8_ENCODING = "UTF-8";
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/util/queue/QueueFileProvider.java]: use the constant from `java.nio.charset.StandardCharsets`

File: core/src/main/java/org/mule/runtime/core/api/store/DeserializationPostInitialisable.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * license, a copy of which has been included with this distribution in the
  * LICENSE.txt file.
  */
-package org.mule.runtime.core.util.store;
+package org.mule.runtime.core.api.store;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/store/DeserializationPostInitialisable.java]: are we okay with this being api? isn't this going to be removed/replaced eventually?

Comment (pablokraan) [inline core/src/main/java/org/mule/runtime/core/api/store/DeserializationPostInitialisable.java]: This is the first iteration of API clean up. This is required to be API on the current state of the code. After the initial clean up, the remaining exported packages will be reviewed again and cleaned if possible

**Pull request merged**